private/bufpkg/bufmodule/bufmodulecache/bufmodulecache_test.go

```diff
@@ -214,6 +220,57 @@ func TestModuleDataProviderBasic(t *testing.T) {
 	)
 }
 
+func TestConcurrentCacheReadWrite(t *testing.T) {
+	t.Skip("skipping expensive test for concurrent cache reads/writes")
```

Comment:
I would expect it to finish a lot faster than this - the first one to acquire should get the write lock and complete writing everything and then the other read locks should read cached data. Are we acquiring the locks in order here?

Reply:
Seems like the major gain/improvement on the test was not calling the delete all on non-existent errors, which makes sense since that is an expensive operation.